packages/peregrine/lib/Router/resolveUnknownRoute.js

```diff
@@ -89,22 +89,23 @@ function remotelyResolveRoute(opts) {
  * @returns {Promise<{type: "PRODUCT" | "CATEGORY" | "CMS_PAGE"}>}
  */
 function fetchRoute(opts) {
+    const query = `query ResolveURL($url: String!) {
```

Comment:
I added a wrapping operation around this (ResolveURL) to use a variable and get rid of the string interpolation, which allows query injection.

Example: venia.magento.com/venia-bottoms.html%22)%20%7B%20type%20id%20%7D%20foo:%20urlResolver(url:%20%22isadora-skirt.html